### ISSUE TITLE
Question - Participation Matrix Softmax

### ISSUE THREAD

**Issue by vopstalf** [Mengqi-Lei/SoftHGNN#1]
Hello,

Thank you for the very nice paper and work.

While reading through the paper and looking through the code, when the softmax normalization of the participation matrix is performed the paper describes this mathematically as a softmax over M the number of hyperedges. This aligns nicely with the idea expressed in the paper of soft-assignment of vertices to hyperedges.
However, when looking at the code the softmax layer seems to be applied across dim =1 , the vertex dimension, unless I'm mistaken.
Am I misunderstanding the dimensions,... is this an intentional change? Did you manage to test performance difference depending on how the softmax is applied? I would've expected F.softmax(logit, dim=2).

Kind regards,

Filip

**Comment by Mengqi-Lei** [comment]
Thank you for your question! We actually considered this issue when designing the model. It corresponds to two different normalization perspectives:

* **Per‐hyperedge normalization**: for each soft hyperedge, the sum of all node participation scores equals 1 (implemented as `softmax(logits, dim=1)`).
* **Per‐node normalization**: for each node, the sum of its participation scores across all soft hyperedges equals 1 (implemented as `softmax(logits, dim=-1)`).

Both approaches are valid, but they can lead to slightly different behaviors in practice depending on the scenario. Here, we recommend first trying the former option, i.e., `softmax(logits, dim=1)`.
